django_celery_monitor/managers.py

```diff
@@ -39,3 +92,23 @@ def purge(self):
                 'DELETE FROM {0.db_table} WHERE hidden=%s'.format(meta),
                 (True, ),
             )
+
+    def update_state(self, state, task_id, defaults):
+        with transaction.atomic():
+            obj, created = self.select_for_update_or_create(
+                task_id=task_id,
+                defaults=defaults,
+            )
+            if created:
+                return obj
+            else:
+                if states.state(state) < states.state(obj.state):
+                    keep = Task.merge_rules[states.RECEIVED]
+                    defaults = dict(
+                        (k, v) for k, v in defaults.items()
+                        if k not in keep
+                    )
+                for k, v in defaults.items():
+                    setattr(obj, k, v)
+                obj.save()
```

Comment:
Care to use `update_fields=tuple(defaults)`? Also, how about this for the else branch in line 104:
```python
if states.state(state) < states.state(obj.state):
    keep = Task.merge_rules[states.RECEIVED]
else:
    keep = {}
for k, v in defaults.items():
    if k not in keep:
        setattr(obj, k, v)
obj.save(update_fields=tuple(defaults.keys()))
```